skia.lua

```diff
@@ -0,0 +1,1021 @@
+    "src/utils/win/SkDWriteGeometrySink.cpp",
+    "src/utils/win/SkHRESULT.cpp",
+  }
+
+  -- Add common Unix
+
+  local common_unix = {
+    "src/ports/SkDebug_stdio.cpp",
+    "src/ports/SkOSFile_posix.cpp",
+    "src/ports/SkTLS_pthread.cpp",
+  }
+
+  -- Add common Linux and Android
+
+  local common_linux = {
+    common_unix,
+    "src/core/SkTSearch.cpp",
+    "src/ports/SkFontMgr_fontconfig.cpp",
+    "src/ports/SkFontMgr_fontconfig_factory.cpp",
+    "src/ports/SkFontConfigInterface_direct.cpp",
+    "src/ports/SkFontHost_FreeType.cpp",
+    "src/ports/SkFontHost_FreeType_common.cpp",
+  }
+
+  local common_android = {
+    common_unix,
+    "src/core/SkTSearch.cpp",
+    "src/ports/SkDebug_android.cpp",
+    "src/ports/SkFontHost_FreeType.cpp",
+    "src/ports/SkFontHost_FreeType_common.cpp",
+    "src/ports/SkFontMgr_android.cpp",
+    "src/ports/SkFontMgr_android_factory.cpp",
+    "src/ports/SkFontMgr_android_parser.cpp",
+  }
+
+  -- add common MacOS and iOS
+
+  local common_cocoa = {
+    common_unix,
+    "src/ports/SkFontHost_mac.cpp",
+    "src/utils/mac/SkStream_mac.cpp",
+  }
+
+  -- Opts SSE
+
+  local opts_sse = {
+    "src/opts/SkOpts_ssse3.cpp",
+    "src/opts/SkOpts_sse41.cpp",
+    "src/opts/SkOpts_sse42.cpp",
+    "src/opts/SkOpts_avx.cpp",
+    "src/opts/SkOpts_crc32.cpp",
+    "src/opts/SkOpts_hsw.cpp",
+  }
+
+  -- Opts arm64
+
+  local opts_arm = {
+    "src/opts/SkOpts_crc32.cpp",
+  }
+  
+  flags {
+    "NoPCH",
+  }
+
+  defines {
+    "SK_SUPPORT_GPU=0",
+    "SK_SCALAR_IS_FLOAT",
+    "SK_HAS_JPEG_LIBRARY",
+    "SK_HAS_PNG_LIBRARY",
+  }
+
+  local t_includedirs = {
+    -- "../giflib/lib",
+    "../libjpeg-turbo",
+    "../libpng",
+    "../zlib",
+    "include",
+    "include/codec",
+    "include/config",
+    "include/core",
+    "include/effects",
+    "include/encode",
+    "include/gpu",
+    "include/images",
+    "include/lazy",
+    "include/pathops",
+    "include/pipe",
+    "include/ports",
+    "include/private",
+    "include/utils",
+    "include/views",
+    "include/xml",
+    "src/codec",
+    "src/core",
+    "src/image",
+    "src/lazy",
+    "src/opts",
+    "src/pdf",
+    "src/ports",
+    "src/sfnt",
+    "src/shaders",
+    "src/utils",
+    "third_party/gif",
+    "third_party/skcms",
+  }
+
+  includedirs { t_includedirs }
+
+  -- -------------------------------------------------------------
+  -- configurations
+  -- -------------------------------------------------------------
+
+  if (os.is("windows") and not _TARGET_IS_WINUWP) then
+    -- -------------------------------------------------------------
+    -- configuration { "windows" }
+    -- -------------------------------------------------------------
+
+    -- common configuration settings
+
+    dofile (_BUILD_DIR .. "/3rdparty_static_win.lua")
+
+    -- project specific configuration settings
+
+    configuration { "windows" }
+
+      defines {
+        "SK_CPU_SSE_LEVEL=20",
```

Comment:
This define can now be moved to SSSE3

Reply:
across all platforms? or just for windows?

Reply:
For all x86 platforms